api/envoy/extensions/filters/network/tcp_proxy/v3/tcp_proxy.proto

```diff
@@ -115,6 +116,9 @@ message TcpProxy {
     // Save the response trailers to the downstream info filter state for consumption
     // by the network filters. The filter state key is ``envoy.tcp_proxy.propagate_response_trailers``.
     bool propagate_response_trailers = 6;
+
+    // Credential injector allows injecting authorization header to upstream proxy.
+    http.credential_injector.v3.CredentialInjector credential_injector = 7;
```

Comment:
it's weird to depend another http filter in the tcp proxy. And obviously the `overwrite` and `allow_request_without_credential` make no sense for tcp proxy. You may only need a credential here.

```
  // The credential to inject into the proxied requests
  // [#extension-category: envoy.http.injected_credentials]
  config.core.v3.TypedExtensionConfig credential = 3 [(validate.rules).message = {required: true}];
```

Reply:
> And obviously the overwrite and allow_request_without_credential make no sense for tcp proxy

You're right.

> You may only need a credential here.

Done.

> it's weird to depend another http filter

I removed the dependency on the filter, and left only the actual injector.